graph/connected_components.cpp

```diff
@@ -51,14 +52,15 @@ void addEdge(std::vector<std::vector<int>> *adj, int u, int v) {
  * @brief Utility function for depth first seach algorithm
  * this function explores the vertex which is passed into.
  *
- * @param u vertex or node to be explored
- * @param visited already visited vertices
+ * @param adj adjacency list of graph.
+ * @param u vertex or node to be explored.
+ * @param visited already visited vertices.
  */
 void explore(const std::vector<std::vector<int>> *adj, int u,
-             std::vector<bool> &visited) {
-    visited[u] = true;
```

Comment:
when a class variable is being passed and the value is not changing, the CPP guidelines suggest that compilers generate faster code when used as constant references
In this case, 
```cpp
void explore(const std::vector<std::vector<int>> *adj, int u,
             const std::vector<bool> &visited) {
```

Reply:
i marked these as const :thinking: , i think they are reverted back in merge conflicts. anyways i will do again.

Reply:
they were marked as constant pointers which are different from constant reference pass.
`const class *var` vs `const class &var` in the function arguments

Reply:
neither is wrong and both are valid per clang-tidy specs we have in place. Just an optimization thing. 
The other errors reported are regarding the use of `uint32_t` type specifiers instead of `unsigned long` and the like.

Reply:
> they were marked as constant pointers which are different from constant reference pass.
> `const class *var` vs `const class &var` in the function arguments


yes got it.


> The other errors reported are regarding the use of uint32_t type specifiers instead of unsigned long and the like.


yes i replaced `long long` with `int64_t`